pkg/api/v1alpha1/cluster_types.go

```diff
@@ -440,9 +442,45 @@ func WorkerNodeGroupConfigurationsSliceEqual(a, b []WorkerNodeGroupConfiguration
 		return false
 	}
 
+	if !WorkerNodeGroupConfigurationsKubeVersionEqual(a, b) {
+		return false
+	}
+
 	return WorkerNodeGroupConfigurationSliceTaintsEqual(a, b) && WorkerNodeGroupConfigurationsLabelsMapEqual(a, b)
 }
 
+// WorkerNodeGroupConfigurationsKubeVersionEqual checks to see if all kubernetesVersion are equal in old and new worker node group conig.
+func WorkerNodeGroupConfigurationsKubeVersionEqual(a, b []WorkerNodeGroupConfiguration) bool {
```

Comment:
from the design, it looks like we're not defaulting(changing) the worker k8s in cluster spec (correct me if im wrong). So we should consider empty value here right?